src/Nox.Types/Common/Enumeration.cs

```diff
@@ -0,0 +1,141 @@
+    public override bool Equals(object obj)
+    {
+        if (obj is not Enumeration<T> otherValue)
+        {
+            return false;
+        }
+
+        var isEqualType = GetType() == obj.GetType();
+        var isEqualId = Id.Equals(otherValue.Id);
+
+        return isEqualType && isEqualId;
+    }
+
+    /// <summary>
+    /// Returns the hash code for the current enumeration.
+    /// </summary>
+    /// <returns>A hash code for the current enumeration.</returns>
+    public override int GetHashCode() => Id.GetHashCode();
+
+    /// <summary>
+    /// Returns a string representation of the current enumeration.
+    /// </summary>
+    /// <returns>A string representation of the current enumeration.</returns>
+    public override string ToString() => Name;
+
+    /// <summary>
+    /// Determines whether two enumerations are equal.
+    /// </summary>
+    /// <param name="a">The first enumeration to compare.</param>
+    /// <param name="b">The second enumeration to compare.</param>
+    /// <returns>True if the enumerations are equal; otherwise, false.</returns>
+    public static bool operator ==(Enumeration<T>? a, Enumeration<T>? b)
+    {
+        if (a is null && b is null)
+            return true;
+
+        if (a is null || b is null)
+            return false;
+
+        return a.Equals(b);
+    }
+
+    /// <summary>
+    /// Determines whether two enumerations are not equal.
+    /// </summary>
+    /// <param name="a">The first enumeration to compare.</param>
+    /// <param name="b">The second enumeration to compare.</param>
+    /// <returns>True if the enumerations are not equal; otherwise, false.</returns>
+    public static bool operator !=(Enumeration<T>? a, Enumeration<T>? b)
+    {
+        return !(a == b);
+    }
+
+    /// <summary>
+    /// Retrieves all instances of the specified enumeration type.
+    /// </summary>
+    /// <typeparam name="TEType">The type of the enumeration.</typeparam>
+    /// <returns>An IEnumerable of the specified enumeration type containing all instances.</returns>
+    public static IEnumerable<TEType> GetAll<TEType>() where TEType : Enumeration<T> =>
+        typeof(TEType).GetFields(BindingFlags.Public |
+                                 BindingFlags.Static |
+                                 BindingFlags.DeclaredOnly)
+            .Select(f => f.GetValue(null))
+            .Cast<TEType>();
+
+    /// <summary>
+    /// Retrieves an instance of the specified enumeration type by its name.
+    /// </summary>
+    /// <typeparam name="TEType">The type of the enumeration.</typeparam>
+    /// <param name="name">The name of the enumeration value.</param>
+    /// <returns>An instance of the specified enumeration type with the given name.</returns>
+    /// <exception cref="InvalidOperationException">Thrown if the name does not match any enumeration value.</exception>
+    public static TEType FromName<TEType>(string name) where TEType : Enumeration<T> =>
```

Comment:
If this method is not used within the Nox Types can we remove it?

Reply:
These methods are created for future usage. Enums should be created by id or name easily. Probably code generator will need them.